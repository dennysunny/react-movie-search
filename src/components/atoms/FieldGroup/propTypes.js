import { string } from 'prop-types';

export const propTypes = {
  id: string,
  label: string,
  help: string,
};

export const defaultProps = {
};
