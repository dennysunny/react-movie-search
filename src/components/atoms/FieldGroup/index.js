import React from 'react';
import {
  FormGroup,
  ControlLabel,
  FormControl,
  HelpBlock,
} from 'react-bootstrap';

import { propTypes, defaultProps } from './propTypes';

const FieldGroup = ({
  id,
  label,
  help,
  ...props
}) => (
    <FormGroup controlId={id}>
      <ControlLabel>{label}</ControlLabel>
      <FormControl {...props} />
      {help && <HelpBlock>{help}</HelpBlock>}
    </FormGroup>
  );

FieldGroup.propTypes = propTypes;
FieldGroup.defaultProps = defaultProps;

export default FieldGroup;
