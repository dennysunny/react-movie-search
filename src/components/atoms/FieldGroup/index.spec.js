import React from 'react';
import { shallow } from 'enzyme';
import FieldGroup from '.';

describe('Atoms/FieldGroup', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<FieldGroup />);

    expect(wrapper).toMatchSnapshot();
  });
});
