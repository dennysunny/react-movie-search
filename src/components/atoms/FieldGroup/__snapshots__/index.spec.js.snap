// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/FieldGroup should render as expected 1`] = `ShallowWrapper {}`;
